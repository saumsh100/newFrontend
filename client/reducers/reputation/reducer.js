import { fromJS } from 'immutable';

import {
  FETCH_REPUTATION,
  FETCH_REPUTATION_SUCCESS
} from './constants';

const initialState = fromJS({
  status: 'loading', // set only to 'loading', 'success' or 'error'
  lastFetched: null,
  data: {
    sourcesNotFound: null,
    sourcesFound: null,
    listingScore: null,
    listingPointScore: {
      industryAverage: null,
      industryLeadersAverage: null,
      pointScore: null
    },
    citationsFound: null,
    sourcesFoundWithErrors: null,
    accuracyScore: null
  }
})

function reputationReducer(state = initialState, action) {
  switch (action.type) {
    case FETCH_REPUTATION:
      return state.set('status', 'loading')
    case FETCH_REPUTATION_SUCCESS:
      return state
        .set('status', 'success')
        .set('lastFetched', (new Date()).toLocaleString())
        .set('data', action.data)
    default:
      return state
  }
}

export default reputationReducer;