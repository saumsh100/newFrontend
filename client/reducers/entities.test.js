
import reducer, {
  createInitialEntitiesState,
  fetchEntities,
  receiveEntities,
  deleteEntity,
  deleteAllEntity,
  updateEntity,
} from './entities';

const initialState = createInitialEntitiesState();

const fetchEntitiesAction = fetchEntities('accounts');
const receiveEntitiesAction = receiveEntities({
  entities: {
    accounts: {
      123: {
        id: '123',
        name: 'test',
      },
    },
  },
});
const deleteEntityAction = deleteEntity({
  key: 'accounts',
  id: '123',
});
const deleteAllEntityAction = deleteAllEntity('accounts');
const updateEntityAction = updateEntity({
  key: 'accounts',
  entity: {
    entities: {
      accounts: {
        123: {
          id: 123,
          name: 'changed name',
        },
      },
    },
  },
});

describe('entities reducer', () => {
  beforeAll(() => {
    jest.spyOn(Date, 'now').mockImplementation(() => new Date(2018, 1, 1));
  });

  test('actions works', () => {
    expect(typeof fetchEntities).toBe('function');
    expect(typeof receiveEntities).toBe('function');
    expect(typeof deleteEntity).toBe('function');
    expect(typeof deleteAllEntity).toBe('function');
    expect(typeof updateEntity).toBe('function');
  });

  test('fetch entitits', () => {
    expect(fetchEntitiesAction).toMatchSnapshot();
  });

  test('receive entitits', () => {
    expect(receiveEntitiesAction).toMatchSnapshot();
  });

  test('delete entitity', () => {
    expect(deleteEntityAction).toMatchSnapshot();
  });

  test('delete all entitits', () => {
    expect(deleteAllEntityAction).toMatchSnapshot();
  });

  test('update entitity', () => {
    expect(updateEntityAction).toMatchSnapshot();
  });

  describe('reducers', () => {
    test('base', () => {
      expect(reducer(initialState, {})).toEqual(initialState);
    });

    test('fetchEntities', () => {
      expect(reducer(initialState, fetchEntitiesAction)).toMatchSnapshot();
    });

    test('receiveEntities', () => {
      expect(reducer(initialState, receiveEntitiesAction)).toMatchSnapshot();
    });

    test('deleteEntity', () => {
      const state = reducer(initialState, receiveEntitiesAction);
      expect(reducer(state, deleteEntityAction)).toMatchSnapshot();
    });

    test('deleteAllEntity', () => {
      const state = reducer(initialState, receiveEntitiesAction);
      expect(reducer(state, deleteAllEntityAction)).toMatchSnapshot();
    });

    test('updateEntity', () => {
      const state = reducer(initialState, receiveEntitiesAction);
      expect(reducer(state, updateEntityAction)).toMatchSnapshot();
    });
  });
});
