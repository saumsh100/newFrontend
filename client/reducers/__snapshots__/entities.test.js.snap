// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entities reducer delete all entitits 1`] = `
Object {
  "payload": "accounts",
  "type": "@entities/DELETE_ALL_ENTITY",
}
`;

exports[`entities reducer delete entitity 1`] = `
Object {
  "payload": Object {
    "id": "123",
    "key": "accounts",
  },
  "type": "@entities/DELETE_ENTITY",
}
`;

exports[`entities reducer fetch entitits 1`] = `
Object {
  "payload": "accounts",
  "type": "@entities/FETCH_ENTITIES",
}
`;

exports[`entities reducer receive entitits 1`] = `
Object {
  "payload": Object {
    "entities": Object {
      "accounts": Object {
        "123": Object {
          "id": "123",
          "name": "test",
        },
      },
    },
  },
  "type": "@entities/RECEIVE_ENTITIES",
}
`;

exports[`entities reducer reducers deleteAllEntity 1`] = `
Immutable.Map {
  "segments": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "reminders": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "recalls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "addresses": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "patientUsers": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "permissions": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "requests": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "accounts": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "users": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "sentReminders": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "sentRecalls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "invites": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "dialogs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "patients": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "chairs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "timeOffs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "enterprises": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "appointments": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "practitionerRecurringTimeOffs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "calls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "practitioners": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "events": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "textMessages": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "availabilities": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "chats": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "weeklySchedules": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "waitSpots": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "services": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
}
`;

exports[`entities reducer reducers deleteEntity 1`] = `
Immutable.Map {
  "segments": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "reminders": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "recalls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "addresses": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "patientUsers": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "permissions": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "requests": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "accounts": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "users": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "sentReminders": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "sentRecalls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "invites": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "dialogs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "patients": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "chairs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "timeOffs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "enterprises": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "appointments": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "practitionerRecurringTimeOffs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "calls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "practitioners": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "events": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "textMessages": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "availabilities": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "chats": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "weeklySchedules": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "waitSpots": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "services": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
}
`;

exports[`entities reducer reducers fetchEntities 1`] = `
Immutable.Map {
  "segments": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "reminders": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "recalls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "addresses": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "patientUsers": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "permissions": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "requests": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "accounts": Immutable.Record {
    "lastUpdated": null,
    "isFetching": true,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "users": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "sentReminders": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "sentRecalls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "invites": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "dialogs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "patients": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "chairs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "timeOffs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "enterprises": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "appointments": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "practitionerRecurringTimeOffs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "calls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "practitioners": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "events": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "textMessages": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "availabilities": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "chats": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "weeklySchedules": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "waitSpots": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "services": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
}
`;

exports[`entities reducer reducers receiveEntities 1`] = `
Immutable.Map {
  "segments": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "reminders": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "recalls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "addresses": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "patientUsers": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "permissions": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "requests": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "accounts": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {
      "123": Immutable.Record {
        "id": "123",
        "addressId": null,
        "name": "test",
        "street": null,
        "logo": null,
        "fullLogoUrl": null,
        "country": null,
        "state": null,
        "city": null,
        "cellPhoneNumberFallback": null,
        "timeInterval": null,
        "zipCode": null,
        "vendastaId": null,
        "phoneNumber": null,
        "timezone": null,
        "contactEmail": null,
        "website": null,
        "twilioPhoneNumber": null,
        "bookingWidgetPrimaryColor": null,
        "destinationPhoneNumber": null,
        "enterpriseId": null,
        "weeklyScheduleId": null,
        "unit": null,
        "canSendReminders": null,
        "canSendRecalls": null,
        "canSendReviews": null,
        "lastSyncDate": "2016-06-01T06:00:00.000Z",
        "googlePlaceId": null,
        "facebookUrl": null,
        "vendastaMsId": null,
        "sendRequestEmail": null,
        "vendastaAccountId": null,
        "vendastaSrId": null,
        "callrailId": null,
        "recallInterval": null,
        "hygieneInterval": null,
        "recallBuffer": null,
        "recallStartTime": null,
        "recallEndTime": null,
        "reviewsInterval": null,
        "massOnlineEmailSentDate": null,
        "suggestedChairId": null,
        "sendUnconfirmedReviews": null,
        "omitChairIds": Array [],
        "omitPractitionerIds": Array [],
        "lastUpdated": 2018-02-01T07:00:00.000Z,
        "isFetching": false,
      },
    },
  },
  "users": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "sentReminders": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "sentRecalls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "invites": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "dialogs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "patients": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "chairs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "timeOffs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "enterprises": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "appointments": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "practitionerRecurringTimeOffs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "calls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "practitioners": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "events": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "textMessages": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "availabilities": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "chats": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "weeklySchedules": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "waitSpots": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "services": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
}
`;

exports[`entities reducer reducers updateEntity 1`] = `
Immutable.Map {
  "segments": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "reminders": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "recalls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "addresses": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "patientUsers": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "permissions": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "requests": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "accounts": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {
      "123": Immutable.Record {
        "id": 123,
        "addressId": null,
        "name": "changed name",
        "street": null,
        "logo": null,
        "fullLogoUrl": null,
        "country": null,
        "state": null,
        "city": null,
        "cellPhoneNumberFallback": null,
        "timeInterval": null,
        "zipCode": null,
        "vendastaId": null,
        "phoneNumber": null,
        "timezone": null,
        "contactEmail": null,
        "website": null,
        "twilioPhoneNumber": null,
        "bookingWidgetPrimaryColor": null,
        "destinationPhoneNumber": null,
        "enterpriseId": null,
        "weeklyScheduleId": null,
        "unit": null,
        "canSendReminders": null,
        "canSendRecalls": null,
        "canSendReviews": null,
        "lastSyncDate": "2016-06-01T06:00:00.000Z",
        "googlePlaceId": null,
        "facebookUrl": null,
        "vendastaMsId": null,
        "sendRequestEmail": null,
        "vendastaAccountId": null,
        "vendastaSrId": null,
        "callrailId": null,
        "recallInterval": null,
        "hygieneInterval": null,
        "recallBuffer": null,
        "recallStartTime": null,
        "recallEndTime": null,
        "reviewsInterval": null,
        "massOnlineEmailSentDate": null,
        "suggestedChairId": null,
        "sendUnconfirmedReviews": null,
        "omitChairIds": Array [],
        "omitPractitionerIds": Array [],
        "lastUpdated": 2018-02-01T07:00:00.000Z,
        "isFetching": false,
      },
    },
  },
  "users": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "sentReminders": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "sentRecalls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "invites": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "dialogs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "patients": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "chairs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "timeOffs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "enterprises": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "appointments": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "practitionerRecurringTimeOffs": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "calls": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "practitioners": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "events": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "textMessages": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "availabilities": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "chats": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "weeklySchedules": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "waitSpots": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
  "services": Immutable.Record {
    "lastUpdated": null,
    "isFetching": false,
    "isCollection": true,
    "models": Immutable.Map {},
  },
}
`;

exports[`entities reducer update entitity 1`] = `
Object {
  "payload": Object {
    "entity": Object {
      "entities": Object {
        "accounts": Object {
          "123": Object {
            "id": 123,
            "name": "changed name",
          },
        },
      },
    },
    "key": "accounts",
  },
  "type": "@entities/UPDATE_ENTITY",
}
`;
